variables.go

```diff
@@ -429,9 +429,16 @@ func lexer(in string) (<-chan token, <-chan error) {
 					lex <- openToken
 					off++
 					varcount++
-				default: // escape any symbol
+				case '$':
+					if content[off+1] == '{' { // remove $
```

Comment:
Is this really safe? The guard before the switch checks only tests for `off >= len(content)`, but not `off + 1`.

Why do we need to check for the pattern `$${` ? The idea was to escape the pattern `$$` -> `$` in general.

Reply:
Yes, indeed. Removing.